.github/workflows/gh-pages.yml

```diff
@@ -39,20 +38,23 @@ jobs:
 
       - name: Build Doxygen
         run: |
-          cd ${GITHUB_WORKSPACE}/gh-pages/doxygen
+          cd doxygen
+          chmod u+x doc-compile.sh
```

Comment:
These shell files can be committed already as executable, which is a thing I'd recommend you do.